ml-algorithms/src/main/java/org/opensearch/ml/engine/encryptor/EncryptorImpl.java

```diff
@@ -94,24 +100,49 @@ public String getMasterKey(String tenantId) {
 
     @Override
     public String encrypt(String plainText, String tenantId) {
-        initMasterKey(tenantId);
-        final AwsCrypto crypto = AwsCrypto.builder().withCommitmentPolicy(CommitmentPolicy.RequireEncryptRequireDecrypt).build();
-        JceMasterKey jceMasterKey = createJceMasterKey(tenantId);
-
-        final CryptoResult<byte[], JceMasterKey> encryptResult = crypto
-            .encryptData(jceMasterKey, plainText.getBytes(StandardCharsets.UTF_8));
-        return Base64.getEncoder().encodeToString(encryptResult.getResult());
+        CountDownLatch latch = new CountDownLatch(1);
+        AtomicReference<Object> encryptResponse = new AtomicReference<>();
+        LatchedActionListener<Boolean> initListener = new LatchedActionListener<>(ActionListener.wrap(result -> {
+            final AwsCrypto crypto = AwsCrypto.builder().withCommitmentPolicy(CommitmentPolicy.RequireEncryptRequireDecrypt).build();
+            JceMasterKey jceMasterKey = createJceMasterKey(tenantId);
+            final CryptoResult<byte[], JceMasterKey> encryptResult = crypto
+                .encryptData(jceMasterKey, plainText.getBytes(StandardCharsets.UTF_8));
+            encryptResponse.set(Base64.getEncoder().encodeToString(encryptResult.getResult()));
+        }, error -> { encryptResponse.set(error); }), latch);
+        initMasterKey(tenantId, initListener);
+        return waitAndGetResult(latch, encryptResponse);
     }
 
     @Override
     public String decrypt(String encryptedText, String tenantId) {
-        initMasterKey(tenantId);
-        final AwsCrypto crypto = AwsCrypto.builder().withCommitmentPolicy(CommitmentPolicy.RequireEncryptRequireDecrypt).build();
-        JceMasterKey jceMasterKey = createJceMasterKey(tenantId);
+        CountDownLatch latch = new CountDownLatch(1);
```

Comment:
We really shouldn't use a latch here, as waiting for it blocks the thread and consumes a thread pool resource.  Try to stick to plain action listeners here.

Reply:
@dbwiddis Removed CountDownLatch completely now.